lib/file_output_handler.py

```diff
@@ -0,0 +1,125 @@
+import os
+
+import six
+
+from lib import utils
+from lib.openapi_final_path_processing import OpenapiPathProcessing
+from lib.swagger_final_path_processing import SwaggerPathProcessing
+
+
+class FileOutputHandler:
+
+    def __init__(self,
+                 rest_package_spec_dict,
+                 api_package_spec_dict,
+                 output_dir,
+                 gen_unique_op_id,
+                 spec,
+                 split_api_rest=False):
+        self.rest_package_spec_dict = rest_package_spec_dict
+        self.api_package_spec_dict = api_package_spec_dict
+        self.output_dir = output_dir
+        self.gen_unique_op_id = gen_unique_op_id
+        self.split_api_rest = split_api_rest
+
+        if spec == '2':
+            self.processor = SwaggerPathProcessing()
+        else:
+            self.processor = OpenapiPathProcessing()
+
+        for package, path_type_tuple in six.iteritems(self.rest_package_spec_dict):
+            self.__preprocess_dict(path_type_tuple[0], path_type_tuple[1])
+        for package, path_type_tuple in six.iteritems(self.api_package_spec_dict):
+            self.__preprocess_dict(path_type_tuple[0], path_type_tuple[1], True)
+
+    def __preprocess_dict(self, path_dict, type_dict, add_camel_case=False):
```

Comment:
Aren't the two underscores "reserved" for built-in functions? Pardon my ignorance, what is their meaning?

Reply:
I think you are referring to __var__ (leading and trailing underscores - ex. the  __init__ function) - they have a special use in python and it is not recommended to name variables or methods in such a way, because there might be collisions with existing or future ones. 

The leading double underscores without trailing ones are used for creating private variables or methods - https://docs.python.org/3/tutorial/classes.html#tut-private, when the python interpreter detects leading underscores it renames the variable or method (referred as name mangling), in order for the subclasses not to override it.

Reply:
I see, thanks for explaining to me! I knew about single leading _ , but I see there is double such, which either superseded it or is used in different context (member functions in a class).